File: PersonalProjects/prosjekt-3/data/scraper/actor.py


from uuid import uuid4


class Actor:
    actors = {}

    def __init__(self):
        pass

    def getActor(self, actor):
        if (actor in Actor.actors.keys()):
            return Actor.actors[actor]
        else:
            id = uuid4()
            Actor.actors[actor] = id
            return id

    def getActors(self, actor):
        return [Actor.getActor(self, a.strip()) for a in actor.split(",")]

File: PersonalProjects/prosjekt-3/backend/crud/actor.ts
import { Client, QueryResult, QueryResultRow } from "pg";

import { Actor } from "../types";

const getExtras = async (result: QueryResult) => {
  for (let i = 0; i < result.rows.length; i++) {
    const row = result.rows[i];
    row.image = process.env.STATIC_ENDPOINT + `/static/actors/${row.id}.jpg`;
  }
  return result.rows;
};

export const getActors = async (
  row: QueryResultRow,
  client: Client
): Promise<Actor[]> => {
  const result: QueryResult = await client.query(
    "select a.name, a.id from movieactor ma inner join actor a on a.id = ma.actorid where ma.movieid = $1::uuid",
    [row.id]
  );
  const rows = await getExtras(result);
  return rows;
};
